platform/mds/mds/src/main/java/org/motechproject/mds/service/DefaultMotechDataService.java

```diff
@@ -414,6 +415,15 @@ public T findById(Long id) {
     }
 
     @Override
+    public Collection<T> findByIds(Collection<Long> ids) {
+        if (ids == null) {
+            return null;
```

Comment:
better return an empty list or throw illegalArgumentException, also any reason for this returning collection since our other methods return lists?